set.mm

```diff
@@ -356721,6 +356885,2332 @@ valuation of the variables (v_0 ` = ( S `` (/) ) ` , v_1 ` = ( S `` 1o
+  df-mvar $a |- mVR = Slot 2 $.
+
+  $( Define the type function in a Metamath formal system.
+     (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+  df-mty $a |- mType = Slot 3 $.
+
+  $( Define the set of typecodes in a Metamath formal system.
+     (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+  df-mtc $a |- mTC = Slot 4 $.
+
+  $( Define the set of axioms in a Metamath formal system.
+     (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+  df-mmax $a |- mAx = Slot 5 $.
+
+  $( Define the set of variable typecodes in a Metamath formal system.
+     (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+  df-mvt $a |- mVT = ( t e. _V |-> ran ( mType ` t ) ) $.
+
+  $( Define the set of "raw expressions", which are expressions without a
+     typecode attached.
+     (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+  df-mrex $a |- mREx = ( t e. _V |-> Word ( ( mCN ` t ) u. ( mVR ` t ) ) ) $.
+
+  $( Define the set of expressions, which are strings of constants and
+     variables headed by a typecode constant.
+     (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+  df-mex $a |- mEx = ( t e. _V |-> ( ( mTC ` t ) X. ( mREx ` t ) ) ) $.
+
+  ${
+    $d a b c d e f h m o p s t v w x y z $.
+    $( Define the set of distinct variable conditions, which are pairs of
+       distinct variables.
+       (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+    df-mdv $a |- mDV = ( t e. _V |->
+      ( ( ( mVR ` t ) X. ( mVR ` t ) ) \ _I ) ) $.
+
+    $( Define the set of variables in an expression.
+       (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+    df-mvrs $a |- mVars = ( t e. _V |-> ( e e. ( mEx ` t ) |->
+        ( ran ( 2nd ` e ) i^i ( mVR ` t ) ) ) ) $.
+
+    $( Define a substitution of raw expressions given a mapping from variables
+       to expressions.
+       (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+    df-mrsub $a |- mRSubst = ( t e. _V |->
+      ( f e. ( ( mREx ` t ) ^pm ( mVR ` t ) ) |-> ( e e. ( mREx ` t ) |->
+        ( ( freeMnd ` ( ( mCN ` t ) u. ( mVR ` t ) ) ) gsum
+    ( ( v e. ( ( mCN ` t ) u. ( mVR ` t ) ) |->
+      if ( v e. dom f , ( f ` v ) , <" v "> ) ) o. e ) ) ) ) ) $.
+
+    $( Define a substitution of expressions given a mapping from variables to
+       expressions.
+       (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+    df-msub $a |- mSubst = ( t e. _V |->
+      ( f e. ( ( mREx ` t ) ^pm ( mVR ` t ) ) |-> ( e e. ( mEx ` t ) |->
+        <. ( 1st ` e ) , ( ( ( mRSubst ` t ) ` f ) ` ( 2nd ` e ) ) >. ) ) ) $.
+
+    $( Define the mapping from variables to their variable hypothesis.
+       (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+    df-mvh $a |- mVH = ( t e. _V |->
+      ( v e. ( mVR ` t ) |-> <. ( ( mType ` t ) ` v ) , <" v "> >. ) ) $.
+
+    $( Define the set of all pre-statements.
+       (Contributed by Mario Carneiro, 14-Jul-2016.) $)
+    df-mpst $a |- mPreSt = ( t e. _V |-> ( ( { d e. ~P ( mDV ` t ) | `' d = d }
+      X. ( ~P ( mEx ` t ) i^i Fin ) ) X. ( mEx ` t ) ) ) $.
```

Comment:
Instead of `'' d = d `, i.e. `d` is reflexive, should this not be something like ` d Er ( mVR ' t ) ` , i.e. `d` is an equivalence relation?

Reply:
`` `' d = d`` says that `d` is symmetric, not reflexive. In fact, I think it should normally be irreflexive, and it's definitely not an equivalence relation.

Reply:
Oh, right, sorry I mixed them up!
My point was I thought it shall be transitive, so something like ` ( d u. ( _I |' ( mVR ' t ) ) ) Er ( mVR ' t ) ` shall hold.